Xeno ex act refactor

goose4429 opened the pull request:
<!-- ***STOP!***  Read this: If this is not a PR ready for review and merge or WIP, open it as a draft PR, using the arrow next to 'Create Pull Request'>

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
 
Code for xeno handle of explosions was strange, and unconcise at times. This PR makes it more standardized, and very slightly nerfs explosions in one particular case.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

It is now easier to change explosion behavior. 
It is also easier to understand how explosions work.
Finally, slightly nerfs light explosions on benos with no bomb resistance (most benos actually) so that they are in line with other cases in terms of stagger and slowdown.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
refactor: Refactored xeno explosion damage procs
balance: Slowdown from light explosions on weak benos reduced from 4 to 3 to be in line with other cases
balance: Stagger is now always 2 less than slowdown with xeno explosions
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Lumipharon commented:
Won't this require all instances of explosions being called in the code to change to the new format (if the args haven't been explicitly named)?

goose4429 commented:
> Won't this require all instances of explosions being called in the code to change to the new format (if the args haven't been explicitly named)?

1. I haven't change the args format, so I don't see how it would matter
2. This is only handler for explosions for xeno. If, for example, a wall is hit by TAT, wall's own explosion handler will be called, which is also untouched. Only xeno.
3. Only thing I've changed which has global effect is actual numbering of defines. But the only place that mattered is in the code I've refactored here. In all other places, only comparisons with global defines are called.

TiviPlus commented on code/__DEFINES/combat.dm:
```diff
@@ -41,6 +41,6 @@
 
 #define EXPLODE_NONE 0
-#define EXPLODE_DEVASTATE 1
+#define EXPLODE_LIGHT 1
 #define EXPLODE_HEAVY 2
-#define EXPLODE_LIGHT 3
+#define EXPLODE_DEVASTATE 3
```
This is tg side so for us to do this it should be done on tg first

goose4429 replied:
Done

TiviPlus commented on code/__DEFINES/combat.dm:
```diff
@@ -41,6 +41,6 @@
 
 #define EXPLODE_NONE 0
-#define EXPLODE_DEVASTATE 1
-#define EXPLODE_HEAVY 2
 #define EXPLODE_LIGHT 3
```
no reason changes

TiviPlus reviewed (changes requested):
Can you ping me on discord I have issues with the code

TiviPlus commented on code/__DEFINES/combat.dm:
```diff
@@ -41,6 +41,6 @@
 #define EXPLODE_NONE 0
-#define EXPLODE_DEVASTATE 1
-#define EXPLODE_HEAVY 2
 #define EXPLODE_LIGHT 3
+#define EXPLODE_HEAVY 2
+#define EXPLODE_DEVASTATE 1
```
```suggestion
#define EXPLODE_DEVASTATE 1
#define EXPLODE_HEAVY 2
#define EXPLODE_LIGHT 3
```
